### PR TITLE
Automender Tweaks

### PR DESCRIPTION
**Pull request by Kiyahitayika**
## What Does This PR Do
Adds new message for when an automender is out of reagents.
Adds visual (colorful and incremental) indicators for mender fullness.

## Why It's Good For The Game
Now you know if the mender runs out or your patient is done healing.
Is your automender running low on reagents? Instead of examining it, you can now see it on the sprite.

## Images of changes
Footage: https://imgur.com/a/bkap41X
Sprites: 
image

The last one blinks!

## Changelog
:cl:
add: New indicators for Automenders - are they more than 50% full? less than 50% full? Empty?
tweak: Automenders now tell you if they power down due to finishing healing OR due to running out of healing reagent.
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
The instantaneous application is *so* inconsequential as to be not worth mentioning; it's not really going to save your life or make much of a difference in the times it matters. In the long drawn out circumstances that I'm sure some hypothesize regarding this, you had time to stop and apply a full amount *anyway*.

I'd prefer to keep it intact

**Comment by Carthusia** [comment]
I like the idea of a changing  color system to indicate how full a mentor is, as currently when a burn mendor runs out, it gets the exact same cross color as a brute mendor, and without closer examination you can end up keeping in your aid kits empty gear.

That being said, the one in the gifs i have a really hard time noticing due to how small it is, perhaps change it to the cross color itself? Green full, Blue half full, flashing color when it is out.

**Comment by Kiyahitayika** [comment]
> The instantaneous application is _so_ inconsequential as to be not worth mentioning; it's not really going to save your life or make much of a difference in the times it matters. In the long drawn out circumstances that I'm sure some hypothesize regarding this, you had time to stop and apply a full amount _anyway_.
> 
> I'd prefer to keep it intact

Thank you for the input! We were talking about this on Discord and I was asked to remove it. I'd be happy to add it back in.

**Comment by Kiyahitayika** [comment]
> I like the idea of a changing color system to indicate how full a mender is, as currently when a burn mendor runs out, it gets the exact same cross color as a brute mendor, and without closer examination you can end up keeping in your aid kits empty gear.
> 
> That being said, the one in the gifs i have a really hard time noticing due to how small it is, perhaps change it to the cross color itself? Green full, Blue half full, flashing color when it is out.

You and I talked in DMs on Discord about this. During our conversation, I experimented with an implementation ala Kugamo's Botany Jugs in PR https://github.com/ParadiseSS13/Paradise/pull/14212. 

The results were not super appealing:
image
image
image

I am definitely interested in workshopping the indicators and figuring out what's best for players :)

**Review by variableundefined — APPROVED** [review]
Approved for code and design

File: code/modules/reagents/reagent_containers/applicator.dm [hunk 1]
```diff
@@ -41,11 +41,20 @@
 
 /obj/item/reagent_containers/applicator/update_icon()
 	cut_overlays()
-
 	if(reagents.total_volume)
 		var/mutable_appearance/filling = mutable_appearance('icons/goonstation/objects/objects.dmi', "mender-fluid")
 		filling.color = mix_color_from_reagents(reagents.reagent_list)
 		add_overlay(filling)
+	var/reag_pct = round((reagents.total_volume / volume) * 100)
+	var/image/light = image('icons/goonstation/objects/objects.dmi', "app_e")
```
Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/applicator.dm:49]: Need to do this the proper way with mutable appearances.

**Pull request merged**